kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/jms/JmsTransactionsIntegrationTest.java

```diff
@@ -0,0 +1,165 @@
+        TransactionManagerServices.getConfiguration().setJournal("null").setGracefulShutdownInterval(2);
+        transaction = TransactionManagerServices.getTransactionManager();
+
+        KieServicesConfiguration jmsConfiguration = createKieServicesJmsConfiguration();
+        KieServerXAConnectionFactory.connectionFactory = jmsConfiguration.getConnectionFactory();
+
+        connectionFactory = new PoolingConnectionFactory();
+        connectionFactory.setClassName("org.kie.server.integrationtests.jbpm.jms.KieServerXAConnectionFactory");
+        connectionFactory.setUniqueName("cf");
+        connectionFactory.setMaxPoolSize(5);
+        connectionFactory.setAllowLocalTransactions(true);
+        connectionFactory.init();
+        jmsConfiguration.setConnectionFactory(connectionFactory);
+
+        jmsConfiguration.setCapabilities(capabilities);
+        jmsConfiguration.setJmsTransactional(true);
+        jmsConfiguration.setResponseHandler(responseHandler);
+
+        KieServicesClient kieServicesClient = KieServicesFactory.newKieServicesClient(jmsConfiguration);
+        return kieServicesClient.getServicesClient(ProcessServicesClient.class);
+    }
+
+    @After
+    public void releaseResources() {
+        if (connectionFactory != null) {
+            connectionFactory.close();
+        }
+        TransactionManagerServices.getTransactionManager().shutdown();
+    }
+
+    @Test
+    public void testTransactionCommit() throws Exception {
+        List<String> capabilities = new ArrayList<String>();
+        capabilities.add(KieServerConstants.CAPABILITY_BPM);
+        ProcessServicesClient transactionalProcessClient = createTransactionalProcessClient(capabilities);
+
+        transaction.begin();
+
+        Long processInstanceId = transactionalProcessClient.startProcess(CONTAINER_ID, PROCESS_ID_USERTASK);
+        assertThat(processInstanceId).isNull();
+
+        transaction.commit();
+
+        List<ProcessInstance> processInstances = queryClient.findProcessInstances(0, 10);
+        assertThat(processInstances).isNotNull().hasSize(1);
+        assertThat(processInstances.get(0).getProcessId()).isEqualTo(PROCESS_ID_USERTASK);
+    }
+
+    @Test
+    public void testTransactionRollback() throws Exception {
+        List<String> capabilities = new ArrayList<String>();
+        capabilities.add(KieServerConstants.CAPABILITY_BPM);
+        ProcessServicesClient transactionalProcessClient = createTransactionalProcessClient(capabilities);
+
+        transaction.begin();
+
+        Long processInstanceId = transactionalProcessClient.startProcess(CONTAINER_ID, PROCESS_ID_USERTASK);
+        assertThat(processInstanceId).isNull();
+
+        transaction.rollback();
+
+        List<ProcessInstance> processInstances = queryClient.findProcessInstances(0, 10);
+        assertThat(processInstances).isNotNull().isEmpty();
+    }
+
+    @Test
+    @Ignore
+    public void testTransactionCommitWithoutSettingCapabilities() throws Exception {
```

Comment:
I would remove this test as it does not provide much of a value. Usually use case is that KieServicesClinet (that provides management capabilities) should be used in default request/reply way why the additional handlers are more for rules, processes etc.